### PR TITLE
Fixed vulnerabilities

### PR DESCRIPTION
**Pull request by lunny**
This issue is reported by gogits/gogs#3959.

### CODE REVIEW

**Comment by tboerger** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Comment by thibaultmeyer** [comment]
LGTM

File: models/token.go [hunk 1]
```diff
@@ -88,7 +88,14 @@ func UpdateAccessToken(t *AccessToken) error {
...
-func DeleteAccessTokenByID(id int64) error {
-	_, err := x.Id(id).Delete(new(AccessToken))
-	return err
+func DeleteAccessTokenByID(id, userID int64) error {
+	cnt, err := x.Id(id).Delete(&AccessToken{
+		UID: userID,
+	})
+	if err != nil {
+		return err
+	} else if cnt != 1 {
```
Comment (strk) [inline models/token.go:97]: nitpicking, the else is not needed here

**Review by strk — APPROVED** [review]
LGTM, but I hadn't tested the code

File: models/token.go (hunk 1 above)
Comment (lunny) [reply models/token.go:97]: It will reduce one line code counting. :)

**Pull request merged**